python/pyiceberg/io/fsspec.py

```diff
@@ -0,0 +1,182 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""FileIO implementation for reading and writing table files that uses fsspec compatible filesystems"""
+
+from typing import Dict, Union
+from urllib.parse import urlparse
+
+from pyiceberg.io import FileIO, InputFile, OutputFile
+from pyiceberg.typedef import EMPTY_DICT
+
+
+class FsspecInputFile(InputFile):
+    """An input file implementation for the FsspecFileIO
+
+    Args:
+        location(str): A URI to a file location
+        fs: An fsspec filesystem instance
+    """
+
```

Comment:
_fs: AbstractFileSystem

Reply:
Fixed!

Reply:
Was it? I think what @Fokko wanted was a declaration of the name, like this:

```python
class FsspecInputFile(InputFile):
    """ docs """

    _fs: AbstractFileSystem    # <-- this

    def __init__(self, location: str, fs: AbstractFileSystem):
        ...
```